File: android/DFM/src/main/java/com/dontflymoney/userdata/Authentication.java
package com.dontflymoney.userdata;

import android.content.Context;

public class Authentication
{
	private Context context;
	private SafeTicket safe;

	private static final String spKey = "Ticket";



	public Authentication(Context context)
	{
		this.context = context;
		safe = new SafeTicket(this.context);
	}
	
	public void Set(String ticket)
	{
		String encryptedTicket = safe.Encrypt(ticket);
		SP.setValue(context, spKey, encryptedTicket);
	}
	
	public String Get()
	{
		String encryptedTicket = SP.getValue(context, spKey);
		return safe.Decrypt(encryptedTicket);
	}

	public boolean IsLoggedIn()
	{
		String ticket = Get();

		return ticket != null && !ticket.isEmpty();
	}

	public void Clear()
	{
		Set(null);
	}
	

}

File: android/DFM/src/main/kotlin/com/darakeon/dfm/user/Authentication.kt
package com.darakeon.dfm.user

import android.content.Context

private val spKey = "Ticket"

fun Context.setAuth(ticket: String) {
	val encryptedTicket = encrypt(ticket)
	setValue(spKey, encryptedTicket)
}

fun Context.getAuth(): String {
	val encryptedTicket = getValue(spKey)
	return decrypt(encryptedTicket)
}

fun Context.isLoggedIn(): Boolean = !getAuth().isEmpty()

fun Context.clearAuth() {
	setAuth("")
}

